### DISCUSSION TITLE
getActivePinia was called with no active Pinia. Did you forget to install pinia? const pinia = createPinia() app.use(pinia) This will fail in production.

### DISCUSSION THREAD

**Discussion by FEsoftdev** [nuxt/nuxt#17000] [Questions]
I have nuxt3 app and it uses pinia for nuxt store.
It works fine on `3.0.0-rc.5` but after upgrade to `3.0.0-rc.8`, this error occurs.
`getActivePinia was called with no active Pinia. Did you forget to install pinia? const pinia = createPinia() app.use(pinia) This will fail in production.`
I only changed nuxt version `3.0.0-rc.5` to `3.0.0-rc.8` to upgrade.

package.json
```
{
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview"
  },
  "dependencies": {
    "@nuxtjs/tailwindcss": "^5.3.0",
    "@pinia/nuxt": "^0.3.0",
    "axios": "^0.27.2",
    "nuxt": "3.0.0-rc.8",
    "pinia": "^2.0.16"
  }
}
```

**Comment by ShaggyTech** [comment]
@Ken0803 Are you able to share your stores and how you are using them?

**Comment by marekmis** [comment]
Same here, Pinia is working fine on `3.0.0-rc.6` but after upgrade to `rc.8` I get the same error. I use Pinia stores in middleware.

```
import { useMyStore } from '@/store/myStore'

export default defineNuxtRouteMiddleware(async (to, from) => { 
  if (process.client) {
    const myStore = useMyStore()

```

**Reply by ShaggyTech** [reply]
> Same here, Pinia is working fine on `3.0.0-rc.6` but after upgrade to `rc.8` I get the same error. I use Pinia stores in middleware.
> 
> ```
> import { useMyStore } from '@/store/myStore'
> 
> export default defineNuxtRouteMiddleware(async (to, from) => { 
>   if (process.client) {
>     const myStore = useMyStore()
> ```

Have you tried using `getActivePinia` as an argument for the store function?  I think something like this will work:

```ts
import { getActivePinia } from 'pinia'
import { useMyStore } from '@/store/myStore'

export default defineNuxtRouteMiddleware(async (to, from) => { 
  if (process.client) {
    const myStore = useMyStore(getActivePinia())
```

You can also make your life easier by adding this to your nuxt.config and then you won't have to import `getActivePinia` or other pinia helpers everytime:

```ts
modules: [
    // ...other modules
    [
      '@pinia/nuxt',
      // add autoImports
      {
        autoImports: [
          'acceptHMRUpdate',
          'defineStore',
          'getActivePinia',
          'storeToRefs',
        ],
      },
    ],
  ],
```

**Comment by FEsoftdev** [comment]
Hi, @marekmis
Please try to clean cache using this command.
`npx nuxi-edge@latest cleanup`

And remove node modules and reinstall.
It works fine for me.
If it doesn't work, can you share your package.json file?
